issue 38 + lazy loading

bttb opened the pull request:
Okay, so here are 2 commits,

the 1st fixes issue 38 and should be obvious. Please include that into the original repo.

The 2nd is some kind of clean up combined with a better implementation of lazy load. If this is too much change, then just ignore the 2nd commit. I think it's an improvement.

bttb commented:
PS: This is my first pull request. If I made something wrong or a pull request usually includes more info, please let me know.

dockleryxk commented:
@bttb you did a great job with it. I responded to your email, let me know what you think

bttb commented:
Hope you can see this. New option autoPlayTransition with values 'slide' (default) and 'fade'. Try it in the demo that I changed, too. I removed "loadFirstSlide", but not 100% sure what it did, so please have a look. Where you called loadFirstSlide I now call loadSlide(0).

dockleryxk commented:
@bttb got it, thank you! I am swamped this week and will merge asap

afomichenko commented:
@bttb I think in your changes the onIndexChanged is no longer emitted when the goToSlide is executed.
